### PR TITLE
Refactor darwinia-ecdsa-authority

### PR DESCRIPTION
**Pull request by AurevoirXavier**

### CODE REVIEW

**Comment by AurevoirXavier** [comment]
@jiguantong you can merge this to your local testnet.

**Comment by boundless-forest** [comment]
@hujw77  Please help take a review.

**Review by hujw77 — APPROVED** [review]
LGTM, The upgrade should be included in darwinia2.0.
Because darwinia1.0 is archive.

File: pallet/ecdsa-authority/src/lib.rs [hunk 1]
```diff
@@ -234,6 +224,13 @@ pub mod pallet {
 				Default::default()
 			}
 		}
+
+		fn on_runtime_upgrade() -> Weight {
```
[the remark this answers is not in the document]
Comment (AurevoirXavier) [reply pallet/ecdsa-authority/src/lib.rs:228]: It's a hook.

**Pull request merged**